tests/integration/executor/test_callbacks.py

```diff
@@ -191,6 +198,25 @@ def test_callback_post(tmpdir, capfd, step, action_type):
         assert out == f"step {step!r}\ncallback\n"
 
 
+@pytest.mark.parametrize("step", list(Step))
+def test_callback_post_state_info(tmpdir, step):
+    callbacks.register_post_step(_state_step_callback, step_list=[step])
```

Comment:
I think it would be useful to make sure that the callback was actually called, to prevent the unlikely-but-possible scenario of the code being refactored in a way that makes the callback silently stop being called.
Something like this, with `_state_step_callback()` as an inner function:

```python
def test_callback_post_state_info(tmpdir, step):
    callback_called = [False]

    def _state_step_callback(info: StepInfo) -> bool:
        assert info.state is not None
        assert isinstance(info.state, StepState)
        callback_called[0] = True
        return True

    # ... rest of test
    assert callback_called[0]
```

Reply:
Implemented as suggested.